### ISSUE TITLE
Guidance - Question dropdowns are confusing

### ISSUE THREAD

**Issue by briri** [DMPRoadmap/roadmap#221]
The Template-Phase-Section-Question dropdowns on the Guidance page are confusing. They do not distinguish between different versions of a template and do not distinguish between funder templates and the org's customisation of those templates.

![screen shot 2017-04-03 at 11 01 01 am](https://cloud.githubusercontent.com/assets/1204467/24623415/2f0497e0-185d-11e7-8576-a2c56549bfe6.png)

**Comment by briri** [comment]
The code behind the dropdowns is also generating errors in the logs
```shell
App 11762 stdout: F, [2017-04-03T10:59:31.770282 #11762] FATAL -- : 
App 11762 stdout: NoMethodError (undefined method `versions' for #<Template:0x0055dfe1e58a28>):
App 11762 stdout:   app/controllers/guidances_controller.rb:34:in `update_phases'
App 11762 stdout: 
```

**Comment by vyruss** [comment]
Will a display format like _$TEMPLATE - $ORG cust._ do the trick?
e.g. _Horizon2020 DMP - University of Marbella cust._
And of course you should never see non-current-version stuff.

**Comment by sjDCC** [comment]
Will this dropdown not be disappearing anyhow since we'll no longer have guidance by question, just annotations? Would be better if users could annotate the question directly rather than adding guidance separately via this interface. It's always been difficult for users to find relevant questions by going through the dropdowns. 
Having multiple entries for each template seems to be a bug btw - that's not usual

**Comment by briri** [comment]
I agree. I'll go ahead and remove these dropdowns from the page.

**Comment by briri** [comment]
@stephaniesimms and @sjDCC this fix has been deployed to the server. The dropdowns were removed from the guidance edit/add pages.

**Comment by stephaniesimms** [comment]
This looks good sans dropdowns.

**Comment by sjDCC** [comment]
Reopening this as it can't be tested at all now. Guidance page throwing an error altogether

![capture](https://cloud.githubusercontent.com/assets/6169940/24914105/1fb8c706-1ecb-11e7-9715-5659074d43f0.PNG)

**Comment by briri** [comment]
Missed this when removing the dropdowns from the guidance page. The tests did not catch it because our test data doesn't contain any guidance with a question_id.

I did a quick check and there are 120 records that have the old question_id populated and no tie in to a theme.
```sql
select * from guidances where question_id is not null;
```
@xsrust and @raycarrick-ed - Should we be moving these question based guidances over to the question table during the migration? I'm not sure how best to resolve it though as some of them have a value in question.guidance as well and they do not match. Here's a quick query I threw together to check:
```sql
select q.id, q.guidance guidance_in_question, g.text guidance_in_guidance 
from questions q left join guidances g on q.id = g.question_id 
where g.question_id is not null;
```
I'm guessing customizations and the changes to the way we handle versioning now complicates this

**Comment by raycarrick-ed** [comment]
@xsrust maybe you could check this ... but it looks to me like we have guidances with question_ids set (146 in my current data) but the question_ids don't seem to match to questions. So somewhere in the migration process we've lost that link. It'd be worth figuring out where this link was broken.

What if we concatenated it with question.guidance? It seems the only way to preserve both at the moment.

**Comment by xsrust** [comment]
In the migration we are concatenating this with `question.guidance` but it seems forgetting to throw away the question_id field from the guidance model after the migration.  I think these guidances with no theme and a question_id can be removed with either a rake task or a migration, and the question_id field can be migrated out of the guidance table

**Comment by briri** [comment]
excellent. thanks for researching @xsrust. I've made a note in #170 so that we don't lose sight of this

**Comment by stephaniesimms** [comment]
This seems to be working as expected now on Roadmap stage. Closing out.

**Comment by sjDCC** [comment]
I'm reopening this one briefly so we can discuss the implications of removing the dropdowns to add guidance by question and ensure that we have a solution for existing data added this way (e.g. by Uni of Eastern Finland on the Academy of Finland template).

I discussed this with @vyruss and we agreed to display the guidance by question annotations on the 'edit cutomisation' view of associated funder templates.  I can't see this in a ticket so just want to make sure we don't overlook it.  I think we agreed to display existing guidance as part of MVP 0.3 and enable edit and addition of new question annotations in the next release.

Also, what happens to any "guidance by question" data that has been added to non-funder templates. Can we query to check we don't have anything in this category as technically it was feasible to add data that way I think.
